src/mem/chunkmap.h

```diff
@@ -141,9 +144,23 @@ namespace snmalloc
    * implementation (for example, to move pagemap updates to a different
    * protection domain).
    */
-  template<typename PagemapProvider = GlobalPagemap>
+  template<
+    template<typename>
+    typename PagemapProviderTemplate,
+    template<auto>
+    typename ChunkmapPagemapTemplate>
```

Comment:
We seem to have lost default parameters here.  Are they no longer needed?

Reply:
They can't be provided because we need to forward-declare the template in the PAL (or AAL), and, AIUI, only the first declaration can provide the defaults?